performance_manager/lib/postgres_utils.py

```diff
@@ -352,31 +352,56 @@ def seed_metadata(self) -> None:
 
 
 def get_unprocessed_files(
-    path_contains: str, sql_session: sessionmaker
-) -> Dict[str, Dict[str, List]]:
-    """check metadata table for unprocessed parquet files"""
+    path_contains: str,
+    db_manager: DatabaseManager,
+    file_limit: Optional[int] = None,
+) -> List[Dict[str, List]]:
+    """
+    check metadata table for unprocessed parquet files
+    groups files into batches of similar partition paths
+    sorts partition paths from oldest to most recent
+
+    returns sorted list of path dictionaries with following layout:
+    {
+        "ids": [metadata table ids],
+        "paths": [s3 paths of parquet files that share path]
+    }
+    """
     process_logger = ProcessLogger(
         "get_unprocessed_files", seed_string=path_contains
     )
     process_logger.log_start()
 
-    paths_to_load: Dict[str, Dict[str, List]] = {}
+    paths_to_load: Dict[float, Dict[str, List]] = {}
     try:
         read_md_log = sa.select((MetadataLog.pk_id, MetadataLog.path)).where(
             (MetadataLog.processed == sa.false())
             & (MetadataLog.path.contains(path_contains))
         )
-        with sql_session.begin() as session:
-            for path_id, path in session.execute(read_md_log):
-                path = pathlib.Path(path)
-                if path.parent not in paths_to_load:
-                    paths_to_load[path.parent] = {"ids": [], "paths": []}
-                paths_to_load[path.parent]["ids"].append(path_id)
-                paths_to_load[path.parent]["paths"].append(str(path))
+        for path_record in db_manager.select_as_list(read_md_log):
+            path_id = path_record.get("pk_id")
+            path = str(path_record.get("path"))
+            path_timestamp = get_utc_from_partition_path(path)
+
+            if path_timestamp not in paths_to_load:
+                paths_to_load[path_timestamp] = {"ids": [], "paths": []}
+            paths_to_load[path_timestamp]["ids"].append(path_id)
+            paths_to_load[path_timestamp]["paths"].append(path)
+
+        paths_found = len(paths_to_load)
+        paths_returned = paths_found
+        if file_limit is not None:
+            paths_returned = file_limit
+
+        process_logger.add_metadata(
+            paths_found=paths_found, paths_returned=paths_returned
+        )
 
         process_logger.log_complete()
 
     except Exception as exception:
         process_logger.log_failure(exception)
 
-    return paths_to_load
+    return [
+        paths_to_load[timestamp] for timestamp in sorted(paths_to_load.keys())
+    ][:file_limit]
```

Comment:
Return a sorted list of dictionary objects limited to `file_limit` amount. Default value of `None` for `file_limit` will return all objects in list.